tcmur_cmd_handler.c

```diff
@@ -1396,15 +1396,17 @@ static int xcopy_parse_parameter_list(struct tcmu_device *dev,
 		tcmu_dev_err(xcopy->src_dev,
 			     "src target exceeds last lba %"PRIu64" (lba %"PRIu64", copy len %u\n",
 			     num_lbas, xcopy->src_lba, xcopy->lba_cnt);
-		return TCMU_STS_RANGE;
+		ret = TCMU_STS_RANGE;
+		goto err;
 	}
 
 	num_lbas = tcmu_dev_get_num_lbas(xcopy->dst_dev);
 	if (xcopy->dst_lba + xcopy->lba_cnt > num_lbas) {
 		tcmu_dev_err(xcopy->dst_dev,
 			     "dst target exceeds last lba %"PRIu64" (lba %"PRIu64", copy len %u)\n",
 			     num_lbas, xcopy->dst_lba, xcopy->lba_cnt);
-		return TCMU_STS_RANGE;
+		ret = TCMU_STS_RANGE;
+		goto err;
 	}
 
 	return TCMU_STS_OK;
```

Comment:
You need to free the 'par' before this too.

Reply:
Yes, thanks for your reply.